### PR TITLE
Add documentation for management components upgrade commands

### PR DESCRIPTION
**Pull request by abhay-krishna**
Add documentation for management components upgrade commands.
Fixes: https://github.com/aws/eks-anywhere/issues/8679

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+linkTitle: "anywhere upgrade management-components"
+---
+
+## anywhere upgrade management-components
+
+Upgrade management components in a management cluster
+
+### Synopsis
+
+The term _management components_ encompasses all Kubernetes controllers and their CRDs present in the management cluster that are responsible for reconciling your EKS Anywhere (EKS-A) cluster. This command is specifically designed to facilitate the upgrade of these management components. Post this upgrade, the cluster itself can be upgraded by updating the 'eksaRelease' field in your eksa cluster object.
```
Comment (vivek-koppuru) [inline docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md:12]: Do you know if we have instructions on upgrading by only changing eksarelease?

**Review by vivek-koppuru — APPROVED** [review]

File: docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md (hunk 1 above)
Comment (abhay-krishna) [reply docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md:12]: We have a small mention of it [here](https://anywhere.eks.amazonaws.com/docs/clustermgmt/cluster-upgrades/management-components-upgrade/#upgrade-management-components). I picked this description up from [here](https://github.com/aws/eks-anywhere/blob/a983cc362971a3e264936b7154a6d72c0e61037e/cmd/eksctl-anywhere/cmd/upgrademanagementcomponents.go#L30), but coming to think of it, the field name should be `eksaVersion` here too right?

Comment (abhay-krishna) [reply docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md:12]: The CRD is called `eksareleases.anywhere.eks.amazonaws.com`, but the field is called `eksaVersion`.

Comment (vivek-koppuru) [reply docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md:12]: We should make that change. Let's also add https://anywhere.eks.amazonaws.com/docs/clustermgmt/cluster-upgrades/management-components-upgrade to the see also below?

Comment (abhay-krishna) [reply docs/content/en/docs/reference/eksctl/anywhere_upgrade_management-components.md]: Good idea, updated.

**Pull request merged**